public final class de.fraunhofer.iem.basics.controlFlow.exceptions.KotlinTryCatchFinallyKt extends java.lang.Object
{

    private static final void tryCatchExample1()
    {
        java.io.PrintStream $stack4, $stack6;
        java.lang.NoSuchMethodException $stack3;
        java.lang.NoSuchFieldException $stack5;
        java.lang.String varReplacer118, varReplacer119, varReplacer120, varReplacer121;

     label1:
        nop;

        varReplacer119 = "Ranjith";

        staticinvoke <de.fraunhofer.iem.basics.controlFlow.exceptions.KotlinTryCatchFinallyKt: void checkValid1(java.lang.String)>(varReplacer119);

        varReplacer120 = "ranj345n";

        staticinvoke <de.fraunhofer.iem.basics.controlFlow.exceptions.KotlinTryCatchFinallyKt: void checkValid2(java.lang.String)>(varReplacer120);

     label2:
        goto label5;

     label3:
        $stack5 := @caughtexception;

        $stack6 = <java.lang.System: java.io.PrintStream out>;

        varReplacer121 = "Something went wrong in tryCatchExample1 (NoSuchFieldException)";

        virtualinvoke $stack6.<java.io.PrintStream: void println(java.lang.Object)>(varReplacer121);

        goto label5;

     label4:
        $stack3 := @caughtexception;

        $stack4 = <java.lang.System: java.io.PrintStream out>;

        varReplacer118 = "Something went wrong in tryCatchExample1 (NoSuchMethodException)";

        virtualinvoke $stack4.<java.io.PrintStream: void println(java.lang.Object)>(varReplacer118);

     label5:
        return;

        catch java.lang.NoSuchFieldException from label1 to label2 with label3;
        catch java.lang.NoSuchMethodException from label1 to label2 with label4;
    }

    private static final void tryCatchExample2()
    {
        java.io.PrintStream $stack4;
        java.lang.NoSuchFieldException $stack3;
        java.lang.String varReplacer122, varReplacer123, varReplacer124;

     label1:
        nop;

        varReplacer124 = "Ranjith";

        staticinvoke <de.fraunhofer.iem.basics.controlFlow.exceptions.KotlinTryCatchFinallyKt: void checkValid1(java.lang.String)>(varReplacer124);

        varReplacer122 = "ranj345n";

        staticinvoke <de.fraunhofer.iem.basics.controlFlow.exceptions.KotlinTryCatchFinallyKt: void checkValid1(java.lang.String)>(varReplacer122);

     label2:
        goto label4;

     label3:
        $stack3 := @caughtexception;

        $stack4 = <java.lang.System: java.io.PrintStream out>;

        varReplacer123 = "Something went wrong in tryCatchExample2 (NoSuchFieldException)";

        virtualinvoke $stack4.<java.io.PrintStream: void println(java.lang.Object)>(varReplacer123);

     label4:
        return;

        catch java.lang.NoSuchFieldException from label1 to label2 with label3;
    }

    private static final void tryCatchExample3()
    {
        java.lang.Throwable $stack4;
        java.lang.NoSuchMethodException $stack6;
        java.lang.NoSuchFieldException $stack9;
        java.lang.String varReplacer125, varReplacer126, varReplacer127, varReplacer128, varReplacer129, varReplacer130, varReplacer131, varReplacer132;
        java.io.PrintStream $stack3, $stack5, $stack7, $stack8, $stack10, $stack11;

     label1:
        nop;

        varReplacer131 = "Ranjith";

        staticinvoke <de.fraunhofer.iem.basics.controlFlow.exceptions.KotlinTryCatchFinallyKt: void checkValid1(java.lang.String)>(varReplacer131);

        varReplacer129 = "ranj345n";

        staticinvoke <de.fraunhofer.iem.basics.controlFlow.exceptions.KotlinTryCatchFinallyKt: void checkValid2(java.lang.String)>(varReplacer129);

     label2:
        $stack3 = <java.lang.System: java.io.PrintStream out>;

        varReplacer132 = "Finally executed in tryCatchExample3";

        virtualinvoke $stack3.<java.io.PrintStream: void println(java.lang.Object)>(varReplacer132);

        goto label9;

     label3:
        $stack9 := @caughtexception;

        $stack10 = <java.lang.System: java.io.PrintStream out>;

        varReplacer125 = "Something went wrong in tryCatchExample3 (NoSuchFieldException)";

        virtualinvoke $stack10.<java.io.PrintStream: void println(java.lang.Object)>(varReplacer125);

     label4:
        $stack11 = <java.lang.System: java.io.PrintStream out>;

        varReplacer126 = "Finally executed in tryCatchExample3";

        virtualinvoke $stack11.<java.io.PrintStream: void println(java.lang.Object)>(varReplacer126);

        goto label9;

     label5:
        $stack6 := @caughtexception;

        $stack7 = <java.lang.System: java.io.PrintStream out>;

        varReplacer128 = "Something went wrong in tryCatchExample3 (NoSuchMethodException)";

        virtualinvoke $stack7.<java.io.PrintStream: void println(java.lang.Object)>(varReplacer128);

     label6:
        $stack8 = <java.lang.System: java.io.PrintStream out>;

        varReplacer130 = "Finally executed in tryCatchExample3";

        virtualinvoke $stack8.<java.io.PrintStream: void println(java.lang.Object)>(varReplacer130);

        goto label9;

     label7:
        $stack4 := @caughtexception;

     label8:
        $stack5 = <java.lang.System: java.io.PrintStream out>;

        varReplacer127 = "Finally executed in tryCatchExample3";

        virtualinvoke $stack5.<java.io.PrintStream: void println(java.lang.Object)>(varReplacer127);

        throw $stack4;

     label9:
        return;

        catch java.lang.NoSuchFieldException from label1 to label2 with label3;
        catch java.lang.NoSuchMethodException from label1 to label2 with label5;
        catch java.lang.Throwable from label1 to label2 with label7;
        catch java.lang.Throwable from label3 to label4 with label7;
        catch java.lang.Throwable from label5 to label6 with label7;
        catch java.lang.Throwable from label7 to label8 with label7;
    }

    private static final void tryCatchExample4()
    {
        java.io.PrintStream $stack3, $stack5;
        java.lang.Throwable $stack4;
        java.lang.String varReplacer133, varReplacer134, varReplacer135, varReplacer136;

     label1:
        nop;

        varReplacer134 = "Ranjith";

        staticinvoke <de.fraunhofer.iem.basics.controlFlow.exceptions.KotlinTryCatchFinallyKt: void checkValid1(java.lang.String)>(varReplacer134);

        varReplacer136 = "ranj345n";

        staticinvoke <de.fraunhofer.iem.basics.controlFlow.exceptions.KotlinTryCatchFinallyKt: void checkValid2(java.lang.String)>(varReplacer136);

     label2:
        $stack3 = <java.lang.System: java.io.PrintStream out>;

        varReplacer133 = "Finally executed in tryCatchExample4";

        virtualinvoke $stack3.<java.io.PrintStream: void println(java.lang.Object)>(varReplacer133);

        goto label5;

     label3:
        $stack4 := @caughtexception;

     label4:
        $stack5 = <java.lang.System: java.io.PrintStream out>;

        varReplacer135 = "Finally executed in tryCatchExample4";

        virtualinvoke $stack5.<java.io.PrintStream: void println(java.lang.Object)>(varReplacer135);

        throw $stack4;

     label5:
        return;

        catch java.lang.Throwable from label1 to label2 with label3;
        catch java.lang.Throwable from label3 to label4 with label3;
    }

    private static final int tryCatchExample5()
    {
        java.lang.Throwable $stack5;
        java.lang.NoSuchMethodException $stack7;
        short l0;
        java.lang.NoSuchFieldException $stack11;
        java.lang.String varReplacer137, varReplacer138, varReplacer139, varReplacer140, varReplacer141, varReplacer142, varReplacer143, varReplacer144;
        java.io.PrintStream $stack4, $stack6, $stack8, $stack9, $stack12, $stack13;

     label1:
        nop;

        varReplacer137 = "Ranjith";

        staticinvoke <de.fraunhofer.iem.basics.controlFlow.exceptions.KotlinTryCatchFinallyKt: void checkValid1(java.lang.String)>(varReplacer137);

        varReplacer141 = "ranj345n";

        staticinvoke <de.fraunhofer.iem.basics.controlFlow.exceptions.KotlinTryCatchFinallyKt: void checkValid2(java.lang.String)>(varReplacer141);

        l0 = 0;

     label2:
        $stack4 = <java.lang.System: java.io.PrintStream out>;

        varReplacer144 = "Finally executed in tryCatchExample5";

        virtualinvoke $stack4.<java.io.PrintStream: void println(java.lang.Object)>(varReplacer144);

        goto label9;

     label3:
        $stack11 := @caughtexception;

        $stack12 = <java.lang.System: java.io.PrintStream out>;

        varReplacer138 = "Something went wrong in tryCatchExample3 (NoSuchFieldException)";

        virtualinvoke $stack12.<java.io.PrintStream: void println(java.lang.Object)>(varReplacer138);

        l0 = -100;

     label4:
        $stack13 = <java.lang.System: java.io.PrintStream out>;

        varReplacer142 = "Finally executed in tryCatchExample5";

        virtualinvoke $stack13.<java.io.PrintStream: void println(java.lang.Object)>(varReplacer142);

        goto label9;

     label5:
        $stack7 := @caughtexception;

        $stack8 = <java.lang.System: java.io.PrintStream out>;

        varReplacer140 = "Something went wrong in tryCatchExample3 (NoSuchMethodException)";

        virtualinvoke $stack8.<java.io.PrintStream: void println(java.lang.Object)>(varReplacer140);

        l0 = -150;

     label6:
        $stack9 = <java.lang.System: java.io.PrintStream out>;

        varReplacer139 = "Finally executed in tryCatchExample5";

        virtualinvoke $stack9.<java.io.PrintStream: void println(java.lang.Object)>(varReplacer139);

        goto label9;

     label7:
        $stack5 := @caughtexception;

     label8:
        $stack6 = <java.lang.System: java.io.PrintStream out>;

        varReplacer143 = "Finally executed in tryCatchExample5";

        virtualinvoke $stack6.<java.io.PrintStream: void println(java.lang.Object)>(varReplacer143);

        throw $stack5;

     label9:
        return l0;

        catch java.lang.NoSuchFieldException from label1 to label2 with label3;
        catch java.lang.NoSuchMethodException from label1 to label2 with label5;
        catch java.lang.Throwable from label1 to label2 with label7;
        catch java.lang.Throwable from label3 to label4 with label7;
        catch java.lang.Throwable from label5 to label6 with label7;
        catch java.lang.Throwable from label7 to label8 with label7;
    }

    private static final void checkValid1(java.lang.String) throws java.lang.NoSuchFieldException
    {
        java.lang.CharSequence l1;
        java.lang.NoSuchFieldException $stack6;
        java.lang.String name, varReplacer145, varReplacer146;
        kotlin.text.Regex $stack4;
        boolean $stack5;

        nop;

        name := @parameter0: java.lang.String;

        l1 = (java.lang.CharSequence) name;

        $stack4 = new kotlin.text.Regex;

        varReplacer146 = ".*\\d.*";

        specialinvoke $stack4.<kotlin.text.Regex: void <init>(java.lang.String)>(varReplacer146);

        $stack5 = virtualinvoke $stack4.<kotlin.text.Regex: boolean matches(java.lang.CharSequence)>(l1);

        if $stack5 == 0 goto label1;

        nop;

        $stack6 = new java.lang.NoSuchFieldException;

        varReplacer145 = "Invalid name";

        specialinvoke $stack6.<java.lang.NoSuchFieldException: void <init>(java.lang.String)>(varReplacer145);

        throw $stack6;

     label1:
        nop;

        return;
    }

    private static final void checkValid2(java.lang.String) throws java.lang.NoSuchMethodException
    {
        java.lang.CharSequence l1;
        java.lang.NoSuchMethodException $stack6;
        java.lang.String name, varReplacer147, varReplacer148;
        kotlin.text.Regex $stack4;
        boolean $stack5;

        nop;

        name := @parameter0: java.lang.String;

        l1 = (java.lang.CharSequence) name;

        $stack4 = new kotlin.text.Regex;

        varReplacer147 = ".*\\d.*";

        specialinvoke $stack4.<kotlin.text.Regex: void <init>(java.lang.String)>(varReplacer147);

        $stack5 = virtualinvoke $stack4.<kotlin.text.Regex: boolean matches(java.lang.CharSequence)>(l1);

        if $stack5 == 0 goto label1;

        nop;

        $stack6 = new java.lang.NoSuchMethodException;

        varReplacer148 = "Invalid name";

        specialinvoke $stack6.<java.lang.NoSuchMethodException: void <init>(java.lang.String)>(varReplacer148);

        throw $stack6;

     label1:
        nop;

        return;
    }

    public static final void main()
    {
        java.io.PrintStream $stack6;
        int $stack3;
        java.lang.String $stack5, varReplacer149;
        java.lang.Integer $stack4;

        nop;

        staticinvoke <de.fraunhofer.iem.basics.controlFlow.exceptions.KotlinTryCatchFinallyKt: void tryCatchExample1()>();

        staticinvoke <de.fraunhofer.iem.basics.controlFlow.exceptions.KotlinTryCatchFinallyKt: void tryCatchExample2()>();

        staticinvoke <de.fraunhofer.iem.basics.controlFlow.exceptions.KotlinTryCatchFinallyKt: void tryCatchExample3()>();

        $stack3 = staticinvoke <de.fraunhofer.iem.basics.controlFlow.exceptions.KotlinTryCatchFinallyKt: int tryCatchExample5()>();

        $stack4 = staticinvoke <java.lang.Integer: java.lang.Integer valueOf(int)>($stack3);

        varReplacer149 = "tryCatchExample5() return ";

        $stack5 = staticinvoke <kotlin.jvm.internal.Intrinsics: java.lang.String stringPlus(java.lang.String,java.lang.Object)>(varReplacer149, $stack4);

        $stack6 = <java.lang.System: java.io.PrintStream out>;

        virtualinvoke $stack6.<java.io.PrintStream: void println(java.lang.Object)>($stack5);

        staticinvoke <de.fraunhofer.iem.basics.controlFlow.exceptions.KotlinTryCatchFinallyKt: void tryCatchExample4()>();

        return;
    }

    public static void main(java.lang.String[])
    {
        java.lang.String[] args;

        nop;

        args := @parameter0: java.lang.String[];

        staticinvoke <de.fraunhofer.iem.basics.controlFlow.exceptions.KotlinTryCatchFinallyKt: void main()>();

        return;
    }
}
